Intro/README.md

```diff
@@ -0,0 +1,455 @@
+Which means that object names cannot be forged easily, which means that
+they can be used in context where a peer needs certain guarantees, or
+to defeat active attacks.
+
+### Immutability of Object Public Areas
+
+Because the name of an object is a digest of its public area, the public
+area cannot be changed after creating it.  One can delete and then
+recreate an object in order to "change" its public area, but this
+necessarily yields a new name (assuming no digest collisions).
+
+## Key Hierarchies
+
+TPMs have multiple key hierarchies, all rooted in a primary decrypt-only
+asymmetric private key derived from a seed, with arbitrarily complex
+trees of keys below the primary key:
+
+```
+                seed
+                 |
+                 |
+                 v
+     primary key (asymmetric encryption)
+                 |
+                 |
+                 v
+       secondary keys (of any kind)
+                 |
+                 |
+                 v
+                ...
+```
+
+There are three built-in hierarchies:
+
+ - platform hierarchy
+ - endorsement hierarchy
+ - storage hierarchy
+
+of which only the endorsement and storage hierarchies will be of
+interest to most readers.
+
+The endorsement hierarchy is used to authenticate (when needed) that a
+TPM is a legitimate TPM.  The primary endorsement key is known as the EK
+(endorsement key).  Hardware TPMs come with a certificate for the EK
+issued by the TPM's manufacturer.  This EK certificate ("EKcert") can be
+used to authenticate the TPM's legitimacy.  The EK's public key
+("EKpub") can be used to uniquely identify a TPM, and possibly link to
+the platform's, and even the platform's user(s)' identities.
+
+## Key Wrapping and Resource Management
+
+The primary key is always a decrypt-only asymmetric private key, and its
+corresponding public key is therefore encrypt-only.  This is largely
+because of key wrapping, where a symmetric key or asymmetric private key
+is encrypted to a TPM's EKpub so that it can be safely sent to that TPM
+so that that TPM can then decrypt and use that secret.
+
+As well as wrapping secrets by encryption to public keys, TPMs also use
+wrapping in a symmetric key known only to the TPM for the purpose of
+saving keys off the TPM.  This is used for resource management: since
+hardware TPMs have very limited resources, objects need to created or
+loaded, used, then saved off-TPM to make room for other objects to be
+loaded (unless they are not to be used again, then saving them is
+pointless).  Only a TPM that saved an object can load it again, but some
+objects can be exported to other TPMs by encrypting them to their
+destination TPMs' EKpubs.
+
+### Controlling Exportability of Keys
+
+A key that is `fixedTPM` cannot leave the TPM in cleartext.  It can be
+saved off the TPM it resides in, but only that TPM can load it again.
+
+A key that is `fixedParent` cannot be re-parented, though if its parent
```

Comment:
Is the exporting of keys, or at least this parent business important enough for this doc? It's confusing, and doesn't seem like a thing that would be commonly used?

Reply:
Actually, yes, I think so.  Perhaps quite a bit more should be said here about key wrapping.  The important thing is to note that one can have keys that are guaranteed not to leave the TPM in the clear nor exportable to other TPMs or systems in any way.  (A key that is fixed to a TPM can still be saved, but only reloaded into the same TPM.)